go-controller/pkg/node/gateway_init.go

```diff
@@ -212,9 +212,7 @@ func (n *OvnNode) initGateway(subnets []*net.IPNet, nodeAnnotator kube.Annotator
 	// as that option does not add default SNAT rules on the GR and the gatewayReady function checks
 	// those default NAT rules are present
 	if !config.Gateway.DisableSNATMultipleGWs && config.Gateway.Mode != config.GatewayModeLocal {
-		waiter.AddWait(gatewayReady, gw.Init)
-	} else {
-		waiter.AddWait(func() (bool, error) { return true, nil }, gw.Init)
+		waiter.AddWait(gatewayReady, func() error { return nil })
```

Comment:
I think you can remove this (and `startup_waiter.go`) completely.
If we really need to wait for `gatewayReady` anywhere in the code, we can do that with `wait.PollImmediate`.

Reply:
right now we wait for the gatewayReady and the managmentPortReady, both really just wait for some flows to be installed by OVN to say we are ready. We could just add both of those as poll immediate, but I guess the benefit of the startup waiter is it runs n number of things to wait on in parallel. I'm not even sure the gatewayReady is very useful. Maybe we should just do the managementPortReady as the only thing we wait on with a poll Immediate...thoughts?

Reply:
@dave-tucker are you sure? the waiter was added to ensure that the node process only writes out it's CNI config file when both management port and gateway were fully configured, becuase they first have to set node annotations, then wait for master to create OVN logical network objects, then complete some of their own configuration based on that.

Reply:
I am not sure so I could be completely wrong 😄 
if we don't need the callback functions anymore, then we can still wait on those conditions using PollImmediate.
IIRC, the `startup_waiter` stuff waits for all conditions and runs the callbacks before returning.